File: Back/src/main/java/com/gestion/user/models/UserLite.java
package com.gestion.user.models;

import lombok.*;

import java.util.HashSet;
import java.util.Set;


@Getter
@Setter
@ToString
@RequiredArgsConstructor
@EqualsAndHashCode(of = {"id"})
public class UserLite {

    private Long id;

    private String prenom;

    private String nom;

    private String email;

    private Set<Role> roles = new HashSet<>();

    public UserLite(User user){
        this.id = user.getId();
        this.nom = user.getAdherent().getNom();
        this.prenom = user.getAdherent().getPrenom();
        this.email = user.getUsername();
        this.roles = user.getRoles();
    }

}

File: Front/src/app/models/userLite.ts
import { Role } from "./role";

export class UserLite {

  id: number;
  prenom: string;
  nom: string;
  email: string;
  roles: Role[];


  constructor() {
    this.id = 0;
    this.prenom = "";
    this.nom = "";
    this.email = "";
    this.roles = [];
  }
}
